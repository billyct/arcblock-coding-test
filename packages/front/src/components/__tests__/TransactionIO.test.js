import {render} from '@testing-library/react'

jest.mock('../TransactionSpent', () => 'transaction-spent-component')
jest.mock('../Hash', () => 'hash-component')

const TransactionIO = require('../TransactionIO').default

describe.each([
  [
    {
      spent: true,
      addr: 'address 1',
      value:'500000000000'
    }
  ],
  [
    {
      spent: false,
      addr: 'address 2',
      value:'500000000000'
    }
  ]
])('test TransactionIO Component case %#', data => {
  it('should render correct', function () {
    const {container} = render(<TransactionIO data={data}/>)
    expect(container).toMatchSnapshot()
  })
})
