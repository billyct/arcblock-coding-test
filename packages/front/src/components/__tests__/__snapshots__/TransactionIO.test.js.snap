// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TransactionIO Component case 0 should render correct 1`] = `
<div>
  <div
    class="flex justify-between py-2"
  >
    <hash-component>
      address 1
    </hash-component>
    <div
      class="flex"
    >
      <div
        class="mr-2"
      >
        5000.00000000 BTC
      </div>
      <transaction-spent-component
        spent="true"
      />
    </div>
  </div>
</div>
`;

exports[`test TransactionIO Component case 1 should render correct 1`] = `
<div>
  <div
    class="flex justify-between py-2"
  >
    <hash-component>
      address 2
    </hash-component>
    <div
      class="flex"
    >
      <div
        class="mr-2"
      >
        5000.00000000 BTC
      </div>
      <transaction-spent-component
        spent="false"
      />
    </div>
  </div>
</div>
`;
